grails-app/controllers/com/unifina/controller/api/StreamApiController.groovy

```diff
@@ -25,6 +26,7 @@ class StreamApiController {
 	]
 
 	private final SimpleDateFormat iso8601 = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+	private final SimpleDateFormat iso8601cal = new SimpleDateFormat("yyyy-MM-dd")
```

Comment:
We could use this opportunity to make the API better. With dates only, the timezone remains ambiguous (based on server timezone), and there is no way to give a more precise timestamp (including time of day).

In our other API endpoints that accept dates (eg. range resend requests in `data-api`) the API accepts two time formats:
- milliseconds since epoch
- ISO8601 strings, as on line 28 here

Let's use this logic for these endpoints too!

So the helper function would look something like this:

```
private Date parseDate(String input) {
    try {
        // try milliseconds timestamp
        return new Date(Long.parseLong(input));
    } catch (...) {
        // fallback to iso8601 timestamp
        return iso8601.parse(input);
    }
}
```

Please add tests that pass both kinds of timestamps!

If date-only strings like `2019-05-10` won't work anymore, that's ok. The frontend guys can update to adjust. Better to have a unified API.

Reply:
url parameter dates (date, end, start) are now in format:
2019-05-10T11:45:40Z
or
1557488802759 -> milliseconds from epoch